### ISSUE TITLE
Support having one file per relation

### ISSUE THREAD

**Issue by aflatter** [rom-rb/rom#103]
<a href="https://github.com/aflatter"><img src="https://avatars.githubusercontent.com/u/168?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [aflatter](https://github.com/aflatter)**
_Wednesday Jan 07, 2015 at 10:39 GMT_
_Originally opened as https://github.com/rom-rb/rom-yaml/issues/1_

----

It'd be great to support one YAML file per relation too!

**Comment by solnic** [comment]
This is very simple to add. We just need to agree on the interface / options. How about this:

``` ruby
# this assumes file-name == dataset-name
ROM.setup("yaml://path/to/directory", relation_per_file: true)

# or explicit settings dataset => file_name

ROM.setup(uri: 'yaml://path/to/directory', files: {
  users: 'people.yml', tasks: 'todos.yml'
})
```

**Comment by cheba** [comment]
Why don't we have both?
```ruby
# Inferred file names (e.g. "users.yml" for users relation)
ROM.setup(uri: 'yaml://path/to/directory', files: true)

# Explicit file names
ROM.setup(uri: 'yaml://path/to/directory', files: {
  users: 'people.yml', tasks: 'todos.yml'
})
```

**Comment by solnic** [comment]
Right, sorry, I meant both would be available

**Comment by aflatter** [comment]
What's supposed to go into URIs (and query parameters) and what should be part of the options hash?

**Comment by solnic** [comment]
@aflatter up to the adapter author

**Comment by aflatter** [comment]
@solnic so why do we have options at all?

**Comment by solnic** [comment]
@aflatter some drivers support URI + extra options, so does ROM. IIRC sequel supports URI + options and moped (mongo) is the same

**Comment by solnic** [comment]
This is now available in rom-yaml 0.1.0
